app.rb

```diff
@@ -3,42 +3,65 @@
 require 'yaml'
 require 'berbix'
 require 'namey'
+require 'json'
+require 'sinatra/custom_logger'
+require 'logger'
+
+set :logger, Logger.new(STDOUT)
 
 berbix_config = YAML.load(File.read('berbix_config.yaml'))
 
 get '/' do
   @generator = Namey::Generator.new
   @username = @generator.name(:common)
-  @uid = rand(10000000000)
+  
+  @uid = 123
 
+  @status = "unknown"
   client = Berbix::Client.new(
     client_secret: berbix_config['client_secret'],
-    environment: :production
   )
 
-  @transaction_tokens = client.create_transaction(
-    customer_uid: @uid, # ID for the user in client database
-    template_key: berbix_config['template_key'], # Template key for this transaction
-  )
+  # Check if refresh token exists for given UID
+  if File.file?("#{@uid.to_i}_refresh_token.txt")
+    # read the refresh token from disk
+    @refresh_token = File.open("#{@uid.to_i}_refresh_token.txt").read
+    @transaction_tokens = Berbix::Tokens.from_refresh(@refresh_token)
+    @transaction_data = client.fetch_transaction(@transaction_tokens)
+    if @transaction_data && !@transaction_data["action"].nil?
+      logger.info @transaction_data["action"]
+      @status = "completed"
+    else
+      @status = "incomplete"
+    end
```

Comment:
@ericlevine what do you think about taking this approach -- essentially calling `fetch_transaction` to introspect on the status of the transaction?

Reply:
Is your goal with this to figure out whether the transaction has been completed or not? I think knowing what the ultimate purpose is will help with evaluating whether this is the right approach.

Reply:
Yes, the goal is to illustrate how one would introspect on the status of the transaction if they wanted to render something to the user in accordance with their current transaction state.

Reply:
Ultimately want to illustrate the "best practices" here.

Reply:
Chatted with @jnnnthnn and landed on this:
- don't encourage this approach, and instead recommend that customers use webhooks and/or oncomplete callbacks to update the state of the transaction, and make sure that customers update the `completed_at` time for a `refresh_token` in their own database
- encourage customers to rely on the client to determine the state of the transaction on load
- this approach could be useful for a backfill if the webhooks/callback approach failed for some reason